ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/AbstractBlockPropagationManagerTest.java

```diff
@@ -641,22 +641,16 @@ public void shouldRequestLowestAnnouncedPendingBlockParent() {
     // test if block propagation manager can recover if one block is missed
 
     blockchainUtil.importFirstBlocks(2);
-    final List<Block> blocks = blockchainUtil.getBlocks().subList(2, 4);
+    final List<Block> blocks = blockchainUtil.getBlocks().subList(2, 6);
 
     blockPropagationManager.start();
 
     // Create peer and responder
     final RespondingEthPeer peer = EthProtocolManagerTestUtil.createPeer(ethProtocolManager, 0);
     final Responder responder = RespondingEthPeer.blockchainResponder(getFullBlockchain());
 
-    // skip first block then create messages from blocklist
-    blocks.stream()
-        .skip(1)
-        .map(this::createNewBlockHashMessage)
-        .forEach(
-            message -> { // Broadcast new block hash message
-              EthProtocolManagerTestUtil.broadcastMessage(ethProtocolManager, peer, message);
-            });
+    EthProtocolManagerTestUtil.broadcastMessage(
+        ethProtocolManager, peer, createNewBlockHashMessage(blocks.get(3)));
```

Comment:
I might be missing something, but I'm a bit unclear why this test needed to be changed. It looks like it was setting up a pretty realistic situation of having missed one block from a chain.

It probably does make sense to add new tests here to demonstrate the change in functionality - in particular having pending blocks received from multiple forks and checking parents of both forks are requested.

Reply:
Yep, I will add a new test